test/run_tests.py

```diff
@@ -96,68 +112,110 @@ def http_server():
+def log_start(command_list, sauceusername, saucekey):
     command = subprocess.list2cmdline(command_list)
-    for hidden in [env.SAUCE_USERNAME, env.SAUCE_ACCESS_KEY]:
+    for hidden in [sauceusername, saucekey]:
         if hidden:
             command = command.replace(hidden, '*' * len(hidden))
     print()
     print('Starting test execution with command:')
     print(command)
 
 
-def process_output(browser, cli_options):
+def get_sauce_conf(browser, sauceusername, saucekey):
+    if browser == 'chrome' and env.TRAVIS:
+        conf = []
+    else:
+        conf = [
+            '--variable', 'SAUCE_USERNAME:{}'.format(sauceusername),
+            '--variable', 'SAUCE_ACCESS_KEY:{}'.format(saucekey),
+            '--variable',
+            'REMOTE_URL:http://{}:{}@ondemand.saucelabs.com:80/wd/hub'.format(
+                sauceusername, saucekey
+            ),
+            '--variable',
+            'DESIRED_CAPABILITIES:build:{0}-{1},tunnel-identifier:{0}'.format(
+                env.TRAVIS_JOB_NUMBER, browser
+            )
+        ]
+    return conf
+
+
+def process_output(browser, suite):
     print('Verifying results...')
     output = os.path.join(env.RESULTS_DIR, 'output.xml')
     robotstatuschecker.process_output(output, verbose=False)
     options = [opt.format(browser=browser) for opt in REBOT_OPTIONS]
+    if suite:
+        options += ['-s', suite]
     try:
-        rebot_cli(options + cli_options + [output])
+        rebot_cli(options + [output])
     except SystemExit as exit:
         return exit.code
 
 
+def sauce_credentials(sauceusername, saucekey):
+    if env.TRAVIS:
+        username = env.SAUCE_USERNAME
+        key = env.SAUCE_ACCESS_KEY
+    else:
+        username = sauceusername
+        key = saucekey
+    return username, key
+
+
 if __name__ == '__main__':
-    if len(sys.argv) < 3 or '--help' in sys.argv:
-        sys.exit(__doc__)
+    parser = argparse.ArgumentParser(description='Library test runner')
+    parser.add_argument(
+        '--interpreter',
+        default='python',
+        help='Interpreter used run the test'
+    )
+    parser.add_argument('browser', help='Browser used in testing')
+    parser.add_argument(
+        '--suite',
+        '-s',
+        help='Selects the test suites by name.'
+    )
+    parser.add_argument(
+        '--scusername',
+        help='Username to order browser from SaucuLabs'
+    )
+    parser.add_argument(
+        '--sckey',
+        help='Access key to order browser from SaucuLabs'
+    )
+
+    args = parser.parse_args()
+    browser = args.browser.lower().strip()
+    if env.TRAVIS and browser != 'chrome' and env.TRAVIS_EVENT_TYPE != 'cron':
+        print(
+            'Can not run test with browser "{}" from SauceLabs\n'
+            'SauceLabs can be used only when running with cron and from '
+            'Selenium2Library master branch, but your event type '
+            'was "{}"'.format(browser, env.TRAVIS_EVENT_TYPE)
```

Comment:
Yes, when running tests with Travis against PR, then it is only possible to use Chrome. Improved text.